x/mongo/driver/topology/tls_connection_source_1_17.go

```diff
@@ -1,19 +1,24 @@
-// Copyright (C) MongoDB, Inc. 2017-present.
+// Copyright (C) MongoDB, Inc. 2022-present.
 //
 // Licensed under the Apache License, Version 2.0 (the "License"); you may
 // not use this file except in compliance with the License. You may obtain
 // a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// +build go1.17
 
 package topology
 
 import (
+	"context"
 	"crypto/tls"
 	"net"
 )
 
 type tlsConn interface {
 	net.Conn
-	Handshake() error
+
+	// Require HandshakeContext on the interface for Go 1.17 and higher.
+	HandshakeContext(ctx context.Context) error
```

Comment:
I think you need to add `HandshakeContext(context.Context)` to the `hangingTLSConn` type [here](https://github.com/mongodb/mongo-go-driver/blob/106bf014881efff40d4b485f74624aa1df51c664/x/mongo/driver/topology/connection_test.go#L1279) to allow the tests to continue building on Go 1.17+.

Consider updating to Go 1.17 and running
```
go test  -run ^$ go.mongodb.org/mongo-driver/...
```
to make sure all the tests successfully build.

Reply:
Ah great catch; for some reason my linter doesn't see these even though I'm on go1.17.3 locally. I did implement `HandshakeContext` for `hangingTLSConn` but it required another pair of conditionally-compiled files. Let me know what you think.

Reply:
Looks good, as long as it works with Go 1.16 and 1.17+ 👍